### PR TITLE
fixes#573: Update contributing.md

### PR DESCRIPTION
**Pull request by himanshuc3**
File Structure is still to be completed. 
Comments and suggestions would be most welcomed.

### CODE REVIEW

**Comment by catarak** [comment]
this is so awesome and i really appreciate how thorough this is! i will read this closely and add some inline comments ✨

File: contributing.md [hunk 1]
```diff
@@ -1,9 +1,126 @@
-# Contributing to the p5.js web editor
+# Contributing to the p5.js web editor 
 
-We welcome community contributions to the p5.js web editor! You can contribute even if you are not a developer by documenting bugs, proposing new features, and updating documentation.
+Hi! Thanks for wanting to contribute to P5's web editor! You are awesome! 😊
```
Comment (catarak) [inline contributing.md]: I want to keep the emphasis that contributing is not just writing code. Could you change this to "Hello! We welcome community contributions to the p5.js web editor. Contributing takes many forms and doesn't have to be writing code, it can be documenting bugs, proposing new features, and updating documentation."

File: contributing.md [hunk 2]
```diff
@@ -1,9 +1,126 @@
...
 
-When opening an issue, be sure to clearly describe the issue, task, or feature. If it is a bug, try to include steps to reproduce or other contextual information, such as the browser or operating system. 
+This **web editor** for **p5.js**, a JavaScript library with the goal of making coding accessible to **artists**, **designers**, **educators**, and **beginners**, is used to help develop using p5.js library in web browser and thus makes for a good starting point for working with p5.
```
Comment (catarak) [inline contributing.md]: **p5.js** is a JavaScript library with the goal of making coding accessible to **artists**, **designers**, **educators**, and **beginners**. The **p5.js web editor** is an environment to make p5.js sketches without needing to download any software or do any configuration, which makes it a great place to start learning how to code and start learning p5.js.

File: contributing.md [hunk 3]
```diff
@@ -1,9 +1,126 @@
...
+
+Please follow the guidelines mentioned at [CODE OF CONDUCT.md](https://github.com/processing/p5.js-web-editor/blob/master/CODE_OF_CONDUCT.md#p5js-code-of-conduct).
+
+# Technologies Used
+
+ > **MERN stack - MongoDB, Express, React/Redux, and Node**. 
+ 
+ - For a reference to the **file structure format** I am using, please look at the [Mern Starter](https://github.com/Hashnode/mern-starter).
+
+ - This project **does not use CSS Modules, but uses Sass**. [BEM guidelines](http://getbem.com/) are followed.
```
Comment (catarak) [inline contributing.md]: - This project **does not use CSS Modules, but uses Sass**. [BEM guidelines and naming conventions](http://getbem.com/) are followed. For repeat and common styles, write OOSCSS (Object-Oriented SCSS) with placeholders and mixins. For organizing styles, follow the [7-1 Pattern](https://sass-guidelin.es/#the-7-1-pattern) for Sass.

File: contributing.md [hunk 4]
```diff
@@ -1,9 +1,126 @@
...
+ - The ESLint configuration is based on a few popular React/Redux boilerplates. Open to suggestions on this. If in development, **you're getting annoyed with ESLint**, you can **remove** it from `webpack.config.dev.js` in the JavaScript loader, or disable any line from eslint by commenting at the end of the line `// eslint-disable-line`.
+
+# Development Installation
+
+To get started on a patch, first read the instruction from [README.md](https://github.com/processing/p5.js-web-editor#development-installation).
+
+# File Structure
+
+```
+/client
```
Comment (catarak) [inline contributing.md]: this could be a separate PR, but it would make sense to also add the file structure for the API/server side stuff too! it's just MVC

File: contributing.md [hunk 5]
```diff
@@ -1,9 +1,126 @@
...
+```
+
+# New Design
+
+Design proposed and theme changes are present at: [Zeplin](https://scene.zeplin.io/project/55f746c54a02e1e50e0632c3).
+
+# How Can I Contribute?
+
+### First Timers
+For first-time contributors or those who want to start with a small task: [check out our list of good first bugs](https://github.com/processing/p5.js-web-editor/labels/good%20first%20issue). These issues have an assigned mentor to help you out and are great issues for learning about the web editor, its file structure and our development process.
```
Comment (catarak) [inline contributing.md]: There's not an "assigned mentor" for these tasks but someone will definitely help out (probably me since I'm currently the lead maintainer). I think it would also be good to add a sentence in here that's like "It's okay to not know how to fix an issue and feel free to ask questions about to approach the problem! We are all just here to learn and make something awesome."

File: contributing.md [hunk 6]
```diff
@@ -1,9 +1,126 @@
...
+
+### Want something more challenging
+If you're already familiar with the project or would like take on something a little more challenging, please take a look at the [priority: high](https://github.com/processing/p5.js-web-editor/labels/priority%3Ahigh) issues.
+
+### Feature Enhancement
+If you want to like building new things, please take a look at [type: feature](https://github.com/processing/p5.js-web-editor/labels/type%3Afeature).
```
Comment (catarak) [inline contributing.md]: "If you want to work on building new things,"

File: contributing.md (hunk 4 above)
Comment (himanshuc3) [reply contributing.md]: Yes, it is incomplete for now. I am removing File structure for now and would upload it in another PR.

**Comment by catarak** [comment]
this is great! i'm gonna merge this in and future edits can be made with more PR's.
